openeogeotrellis/load_stac.py

```diff
@@ -505,6 +506,25 @@ def intersects_temporally(interval) -> bool:
     return GeopysparkDataCube(pyramid=gps.Pyramid(levels), metadata=metadata)
 
 
+def get_best_url(asset):
+    """
+    Relevant doc: https://github.com/stac-extensions/alternate-assets
+    """
+    alternate = asset.extra_fields.get("alternate")
+    if alternate:
+        for key, alternate_local in alternate.items():
+            if key != "local":
+                continue
+            href = alternate_local.get("href")
+            # Checking if file exists takes around 10ms on /data/MTDA mounted on laptop
+            # Checking if URL exists takes around 100ms on https://services.terrascope.be
+            # Checking if URL exists depends also on what Datasource is used in the scala code.
+            # That would be hacky to predict here.
```

Comment:
@VictorVerhaert Is this logic fine for you?

Reply:
In some collections, the alternate link is not local but 's3', in those cases when we have access to this s3 it should be used.
Note that these s3 alternate links might contain extra fields indicating an endpoint or region.

I might suggest using this package instead of path: https://pypi.org/project/universal-pathlib/ in those cases, which allows the usage of both paths as well as s3 links.

Reply:
S3 would be a different code path. To support it, it would be nice to have an example stac catalog that uses this feature.
The `local` path can refer to a mounted s3 bucket tough, like `/eodata`.

Reply:
It works for mounted s3 buckets now. But I can not test it on `/eodata/Global-Mosaics/` specifically yet. That can be done, when the `eo:bands` property is supported, or when Cloudferro's gives access: `CDSE-1342`